Framework/Geometry/src/Instrument.cpp

```diff
@@ -1235,5 +1237,15 @@ Instrument::ContainsState Instrument::containsRectDetectors() const {
 
 } // containsRectDetectors
 
+/// Only for use by ExperimentInfo. Returns the pointer to the DetectorInfo.
+const Beamline::DetectorInfo *Instrument::detectorInfo() const {
+  return m_detectorInfo;
```

Comment:
Well, it is a pointer so the client (`ExperimentInfo`) can just check if it is null (https://github.com/mantidproject/mantid/pull/18276/files#diff-f92af670836a7a4178efefc827b27f95R159). The null check would be useful if this would return a reference, I guess?